src/apps/_experiments/weather.cpp

```diff
@@ -0,0 +1,797 @@
+                                          thunderstorm_heavy, squall_tornado };
+};
+
+WeatherParser::WeatherParser(){}
+
+string WeatherParser::encodeWeather(DynamicJsonDocument& doc){
+  const char* code = nullptr;
+  code = doc["cod"];
+  if(strcmp("200",code)){
+    if(code==nullptr){
+      Serial.println("Error, corrupted API response.");
+      return  "ERROR_CORRUPTED_RESPONSE";
+    }else{
+      Serial.print("Error, response code: ");
+      Serial.println(code);
+      return "ERROR_API_RESPONSE";
+    }
+  }
+  cnt = doc["cnt"];
+  time_t time = doc["list"][0]["dt"];
+  WeatherEncoder encoder;
+  encoder.setTimestamp(time);
+  bool res;
+  for(int i=0; i<cnt; i++){
+    weather_update_t update;
+    int temp = doc["list"][i]["main"]["temp"];
+    update.temp = temp - 273;
+    update.humidity = doc["list"][i]["main"]["humidity"];
+    int pressure = doc["list"][i]["main"]["pressure"];
+    update.pressure = pressure;
+    update.weather = this->_getWCond(doc["list"][i]["weather"][0]["id"]);
+    res = encoder.setUpdate(update);
+    if (!res){
+      return "ERROR_INPUT";
+    }
+  }
+  return encoder.getEncoded();
+}
+
+int WeatherParser::_getWCond(int weather_code){
+  for(int i=0; i<15; i++){
+    for(int j=0; j < weather_conditions[i].size(); j++){
+      if(weather_code == weather_conditions[i][j]){
+        return i;
+      }
+    }
+  }
+  return 15; // unknown weather def 
+}
+
+
+
+
+void OswAppWeather::drawSun(int x, int y, int radius ){
+  this->hal->gfx()->fillCircle(x, y, radius ,rgb888(255, 250, 1));
+}
+
+
+void OswAppWeather::drawMoon(int x, int y, int radius ){
+    this->hal->gfx()->fillCircle(x, y, radius ,rgb888(255, 255, 255));
+}
+/*draw a cloud given: x coordinate and y coordinate.
+  color: rgb888 color
+  k: size coef. NOTE: if k is > 1 cloud may be partially out of the screen
+*/
+void OswAppWeather::drawCloud(int x, int y, uint32_t color, float k ){
+    x = x+22;
+    y=y+15;
+    float radius = 15*k;
+    float k1=12*k;
+    float j1=5*k;
+    float k2=6*k;
+    float j2=1*k;
+    float k3=14*k;
+   
+    this->hal->gfx()->fillCircle(x, y, radius ,color);
+
+    this->hal->gfx()->fillCircle(x-k1, y+j1, radius-j1  ,color);
+
+    this->hal->gfx()->fillCircle(x-k2, y-j2, radius-j1 ,color);
+
+    this->hal->gfx()->fillCircle(x+k3, y+j1, radius-j1 ,color);
+}
+
+void OswAppWeather::_drawDroplet(int x, int y, uint32_t color){
+  this->hal->gfx()->drawLine(x, y, x+5, y+5, color);
+  this->hal->gfx()->drawLine(x+1, y, x+6, y+5, color);
+  this->hal->gfx()->drawLine(x+2, y, x+7, y+5, color);
+}
+
+void OswAppWeather::drawRain(int x, int y ){
+    this->_drawDroplet(x,y);
+    this->_drawDroplet(x+10,y+4);
+    this->_drawDroplet(x+15, y);
+}
+
+void OswAppWeather::drawSnow(int x, int y, int level){
+  switch (level)
+  {
+  case 1:
+    this->hal->gfx()->fillCircle(x+10,y+3,3,rgb888(255,255 , 255));
+    break;
+  case 2:
+    this->hal->gfx()->fillCircle(x,y+2,3,rgb888(255,255 , 255));
```

Comment:
Why don't you use the prepared ui color?
`ui->getDanger()`

Reply:
because i want the snow to be always white.

Reply:
Do you think that i should create a separate class for the icons ? In this case i will also provide to othe apps the possibility to print the weather icons.

Reply:
I agree. If you make the icon available as a parameter for size (scale) and location, it will be available for notification in other apps.

Reply:
ok 👍